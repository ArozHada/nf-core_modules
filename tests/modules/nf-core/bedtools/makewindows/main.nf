#!/usr/bin/env nextflow

nextflow.enable.dsl = 2
moduleDir = launchDir

include { BEDTOOLS_MAKEWINDOWS } from "$moduleDir/modules/nf-core/bedtools/makewindows/main.nf"

workflow test_bedtools_makewindows {

    input = [
        [ id:'test'],
        file(params.test_data['sarscov2']['genome']['test_bed'], checkIfExists: true)
    ]

    BEDTOOLS_MAKEWINDOWS ( input, true )
}
