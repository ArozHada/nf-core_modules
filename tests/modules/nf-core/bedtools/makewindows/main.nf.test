nextflow_process {

    name "Test Process BEDTOOLS_MAKEWINDOWS"
    script "modules/nf-core/bedtools/makewindows/main.nf"
    process "BEDTOOLS_MAKEWINDOWS"
    config "tests/modules/nf-core/bedtools/makewindows/nextflow.config"
    tag "bedtools_makewindows"

    test("Sarscov2 Genome BED") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                            [ id:'test2'],
                            file(params.test_data['sarscov2']['genome']['test_bed'], checkIfExists: true)
                           ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(params.outdir).list()).match("bed") }
            )
        }
    }

    test("Sarscov2 Genome Fasta") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                            [ id:'test2'],
                            file(params.test_data['sarscov2']['genome']['genome_fasta_fai'], checkIfExists: true)
                           ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(params.outdir).list()).match("fasta") }
            )
        }
    }
}
