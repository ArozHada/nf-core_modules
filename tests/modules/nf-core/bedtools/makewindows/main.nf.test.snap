{
    "bed": {
        "content": [
            [
                "test2.bed:md5,0cf6ed2b6f470cd44a247da74ca4fe4e",
                "versions.yml:md5,121fffc46a67e20ad562690920aef5b3"
            ]
        ],
        "timestamp": "2023-04-11T22:02:06+0000"
    },
    "fasta": {
        "content": [
            [
                "test2.bed:md5,622d1f62786fe4239b76c53168f21c54",
                "versions.yml:md5,121fffc46a67e20ad562690920aef5b3"
            ]
        ],
        "timestamp": "2023-04-11T22:02:06+0000"
    }
}